daemon/daemon.go

```diff
@@ -358,10 +354,12 @@ func (d *Daemon) Start() {
 		case state.RestartDaemon:
 			d.tomb.Kill(nil)
 		case state.RestartSystem:
-			cmd := exec.Command("shutdown", "+10", "-r", shutdownMsg)
-			if out, err := cmd.CombinedOutput(); err != nil {
-				logger.Noticef("%s", osutil.OutputErr(out, err))
+			bootloader, err := partition.FindBootloader()
+			if err != nil {
+				logger.Noticef("cannot get bootloader: %s", err)
+				break
```

Comment:
Shouldn't we have a proper fallback here when no bootloader is found and still call "shutdown"? Not sure if we can otherwise supports environments like docker or LXD.